### PR TITLE
Changes for deployment events:

### PR DESCRIPTION
**Pull request by avasquez614**
* Added processor that saves cache clear/rebuild context events in a file.
* Added originalChangeSet to list of parameters of AbstractMainDeploymentProcessor.doExecute so processors can modify the change set of the deployment.

Ticket craftercms/craftercms#2935

### CODE REVIEW

File: src/main/java/org/craftercms/deployer/impl/processors/FileBasedDeploymentEventProcessor.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package org.craftercms.deployer.impl.processors;
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/deployer/impl/processors/FileBasedDeploymentEventProcessor.java:17]: Missing copyright notice.

File: src/main/java/org/craftercms/deployer/impl/processors/FileBasedDeploymentEventProcessor.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+/**
+ * Triggers a deployment event that consumers of the repository (Crafter Engines) can subscribe to by reading a file
+ * written by this processor.
+ *
+ * @author avasquez
+ */
+public class FileBasedDeploymentEventProcessor extends AbstractMainDeploymentProcessor {
+
+    private static final Logger logger = LoggerFactory.getLogger(FileBasedDeploymentEventProcessor.class);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/deployer/impl/processors/FileBasedDeploymentEventProcessor.java:50]: Line too long.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/deployer/impl/processors/FileBasedDeploymentEventProcessor.java (hunk 2 above)
Comment (avasquez614) [reply src/main/java/org/craftercms/deployer/impl/processors/FileBasedDeploymentEventProcessor.java:50]: Discussed this, won't change the size, we'll look later on a way to expand the GitHub editor.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Missing copyright notice.

**Comment by avasquez614** [comment]
@sumer Not sure why it's not appearing, but the copyright notice is now in the file: https://github.com/craftercms/deployer/blob/f12f2695ba6c580afd2867b46f19764550c0b2b8/src/main/java/org/craftercms/deployer/impl/processors/FileBasedDeploymentEventProcessor.java

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**